---
title: "BeWo PLAC1 KD Bulk RNAseq analyses"
date: "`r Sys.Date()`"
author:
  - name: Arun Seetharam
    affiliation: Tuteja Lab
    affiliation_url: https://www.tutejalab.org
output:
  rmdformats::readthedown:
    self_contained: true
    thumbnails: false
    lightbox: true
    gallery: true
    highlight: tango
---

```{r setup, include=FALSE}
options(max.print = "125")
knitr::opts_chunk$set(
  echo = TRUE,
  collapse = TRUE,
  comment = "#>",
  fig.path = "assets/",
#  dev = 'pdf',
  dpi = 300,
#  fig.ext = "pdf",
  fig.width = 12,
  fig.height = 8.5,
  prompt = FALSE,
  tidy = FALSE,
  message = FALSE,
  warning = TRUE
)
knitr::opts_knit$set(width = 75)
```

# Environment Setup

```{bash eval = FALSE}
salloc -N 1 --exclusive -p amd -t 8:00:00
conda activate star
# working dir
mkdir -p /work/LAS/geetu-lab/arnstrm/BeWo_PLAC1_KD
cd /work/LAS/geetu-lab/arnstrm/BeWo_PLAC1_KD
mkdir -p 1_data
mkdir -p 2_fastqc
mkdir -p 3_STAR-mapping
mkdir -p 4_featureCounts
mkdir -p 5_multiqc
# file structure
tree -L 1
.
├── 1_data
├── 2_fastqc
├── 3_STAR-mapping
├── 4_featureCounts
└── 5_multiqc
```

## Raw data

Raw data was downloaded from the LSS using `rsync` command.

```{bash, eval=FALSE, engine="sh"}
cd 1_data
rsync -avP /lss/folder/path/BeWo_PLAC1_KD ./1_data
# GEO link will be included later
```

## Genome/annotation

Additional files required for the analyses were downloaded from [GenCode](https://www.gencodegenes.org/human/). The downloaded files are as follows:

```{bash eval = FALSE} 
cd 3_STAR-mapping
wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_42/GRCh38.p13.genome.fa.gz
wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_42/gencode.v42.primary_assembly.annotation.gff3.gz
wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_42/gencode.v42.primary_assembly.annotation.gtf.gz
gunzip GRCh38.p13.genome.fa.gz
gunzip gencode.v42.primary_assembly.annotation.gtf.gz
gunzip gencode.v42.primary_assembly.annotation.gff3.gz
# ids for prot coding genes
awk '$3=="gene" && $9 ~/gene_type=protein_coding;/ {
        split($9,a,";"); print gensub(/gene_id=/, "", 1, a[2])
        }'  gencode.v42.primary_assembly.annotation.gff3 > GRCh38.protein_coding
# you should have 20,054 ids (protein-coding genes)
```

## FastQC

Quality inspection of the reads. The `multiqc` report, collating all samples together are provided as html file.

```{bash eval = FALSE}
cd 2_fastqc
for fq in ../1_data/*.fq.gz; do
  fastqc --threads $SLURM_JOB_CPUS_PER_NODE $fq;
done
```


# Mapping

To index the genome, following command was run (in an interactive session).

```{bash eval = FALSE} 
fastaGenome="GRCh38.p13.genome.fa"
gtf="gencode.v42.primary_assembly.annotation.gtf"
STAR --runThreadN $SLURM_JOB_CPUS_PER_NODE \
     --runMode genomeGenerate \
     --genomeDir $(pwd) \
     --genomeFastaFiles $fastaGenome \
     --sjdbGTFfile $gtf \
     --sjdbOverhang 1
```

Each `fastq` file was mapped to the indexed genome as using `runSTAR_map.sh` script shown below:

```{bash eval = FALSE}
#!/bin/bash
conda activate star
index=/work/LAS/geetu-lab/arnstrm/GRCh38_index
read1=$1
read2=$(echo ${read1} | sed 's/_R1_/_R2_/g')
cpus=${SLURM_JOB_CPUS_PER_NODE}
out=$(basename ${read1} | cut -f 1-2 -d "_")
STAR \
--runThreadN ${cpus} \
--genomeDir ${index} \
--outSAMtype BAM SortedByCoordinate \
--quantMode GeneCounts \
--outFilterScoreMinOverLread 0.3 \
--outFilterMatchNminOverLread 0.3 \
--outFileNamePrefix ${out}_ \
--readFilesCommand zcat \
--outWigType bedGraph \
--outWigStrand Unstranded \
--outWigNorm RPM \
--readFilesIn ${read1} ${read2}
```

Mapping was run with a simple loop:

```{bash eval = FALSE}
for fq in *_R1_*fastq.gz; do
  runSTAR_map.sh $fq;
done
```

# Counts

For generating counts from the mapped reads, we used `subread` package program `featureCounts`. All bam files were supplied together to generate a single count file for individual samples.

```{bash eval = FALSE}
cd 3_STAR-mapping
realpath *.bam > ../4_featureCounts/bam.fofn
cd ../4_featureCounts
gtf="gencode.v42.primary_assembly.annotation.gtf"
while read line; do
ln -s $line;
done
featureCounts \
   -T ${SLURM_CPUS_ON_NODE} \
   -a ${gtf} \
   -t exon \
   -g gene_id \
   -p \
   -B \
   --countReadPairs \
   -o merged_counts.txt \
   --tmpDir ./tmp *.bam
```

The generated counts file was processed to use it direclty with `DESeq2`

```{bash eval = FALSE}
grep -v "^#" merged_counts.txt |\
    cut -f 1,7- |\
    sed 's/_Aligned.sortedByCoord.out.bam//g' > merged_clean-counts.txt
head -n 1 merged_clean-counts.txt > header
grep -Fw -f GRCh38.protein_coding merged_clean-counts.txt > body
cat header body > counts_genes.tsv
rm body head
```

Create a info file:

```{bash eval = FALSE}
head -n 1 counts_genes.tsv |\
   tr "\t" "\n" |\
   grep -v "^Geneid" |\
   awk '{print $1"\t"gensub(/.$/, "", 1,$1)}' > info.tsv
```

# Differential expression analysis 

Differential expression (DE) analyses using `DESeq2` was performed as shown below.

## Prerequisites

R packages required for this section are loaded

```{r, warnings=TRUE, message=FALSE}
# set path
setwd("/work/LAS/geetu-lab/arnstrm/BeWo_PLAC1_KD")
# load the modules
library(tidyverse)
library(DESeq2)
library(pheatmap)
library(ggrepel)
library(RColorBrewer)
library(reshape2)
library(TissueEnrich)
library(plotly)
library(cowplot)
library(biomaRt)
library(scales)
library(kableExtra)
library(htmlwidgets)
library(DT)
library(enrichR)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(org.Rn.eg.db)
library(data.table)
library(GOSemSim)
library(DOSE)
library(pathview)
```

## Import datasets

The `counts` data and its associated metadata (`coldata`) are imported for analyses.

```{r datasetS, warnings=TRUE, message=FALSE}
countsFile = 'assets/counts_genes.tsv'
groupFile = 'assets/info.tsv'
coldata <-
  read.csv(
    groupFile,
    row.names = 1,
    sep = "\t",
    header = FALSE,
    stringsAsFactors = TRUE
  )
colnames(coldata) <- "condition"
cts <- as.matrix(read.delim(countsFile, row.names = 1, header = TRUE))
```

Reorder columns of `cts` according to `coldata` rows. Check if samples in both files match.

```{r order2, warnings=TRUE, message=FALSE}
all(rownames(coldata) %in% colnames(cts))
cts <- cts[, rownames(coldata)]
```

## DESeq2

The batch corrected read counts are then used for running DESeq2 analyses

```{r deseq2C, warnings=TRUE, message=FALSE}
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ condition)
vsd <- vst(dds, blind = FALSE)
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep, ]
dds <- DESeq(dds)
```

## PCA plot for QC

PCA plot for the dataset that includes all libraries.

```{r pcaFull_C1-C2, fig.cap="Figure 4: PCA plot for the first 2 principal components", fig.width=8, fig.height=5}
rv <- rowVars(assay(vsd))
select <-
  order(rv, decreasing = TRUE)[seq_len(min(500, length(rv)))]
pca <- prcomp(t(assay(vsd)[select, ]))
percentVar <- pca$sdev ^ 2 / sum(pca$sdev ^ 2)
intgroup = "condition"
intgroup.df <- as.data.frame(colData(vsd)[, intgroup, drop = FALSE])
group <- if (length(intgroup) == 1) {
  factor(apply(intgroup.df, 1, paste, collapse = " : "))
}
d <- data.frame(
  PC1 = pca$x[, 1],
  PC2 = pca$x[, 2],
  intgroup.df,
  name = colnames(vsd)
)
ggplot(d, aes(PC1, PC2, color = condition)) +
  scale_shape_manual(values = 1:12) +
  scale_color_manual(values = c('BeWo_ctr' 		= '#980c80',
                                'BeWo_KD' 	= '#00b462')) +
  theme_bw() +
  theme(legend.title = element_blank()) +
  geom_point(size = 2, stroke = 2) +
  geom_text_repel(aes(label = name)) +
  xlab(paste("PC1", round(percentVar[1] * 100, 2), "% variance")) +
  ylab(paste("PC2", round(percentVar[2] * 100, 2), "% variance"))
```

## Sample distance for QC

```{r distance, fig.cap="Figure 5: Euclidean distance between samples", fig.width=8, fig.height=5}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- colnames(vsd)
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors)
```

## Set contrasts and find DE genes

```{r cotnrasts, warnings=TRUE, message=FALSE}
res.KDvsCT <-
  results(dds,
          contrast = c("condition",
                       "BeWo_KD",
                       "BeWo_ctr"))
```

# Functions

## Import functions for processing and plotting

```{r processDE1, warnings=TRUE, message=FALSE}
source("/work/LAS/geetu-lab/arnstrm/myR_functions/plot_enrichR.R")
source("/work/LAS/geetu-lab/arnstrm/myR_functions/processDE.R")
source("/work/LAS/geetu-lab/arnstrm/myR_functions/rat_id_conversion.R")
source("/work/LAS/geetu-lab/arnstrm/myR_functions/run_pce.R")
source("/work/LAS/geetu-lab/arnstrm/myR_functions/save_pheatmap.R")
source("/work/LAS/geetu-lab/arnstrm/myR_functions/theme_clean.R")
source("/work/LAS/geetu-lab/arnstrm/myR_functions/volcano_plots.R")
```

## Gene information

```{r martObj1, eval = FALSE}
ensembl = useMart("ENSEMBL_MART_ENSEMBL")
listDatasets(ensembl) %>%
  filter(str_detect(description, "Human"))
ensembl = useDataset("hsapiens_gene_ensembl", mart = ensembl)
listFilters(ensembl) %>%
  filter(str_detect(name, "ensembl"))
filterType <- "ensembl_gene_id_version"
head(rownames(counts))
counts <- read.delim("assets/counts_genes.tsv", row.names = 1, header = TRUE)
head(rownames(counts))
filterValues <- rownames(counts)
listAttributes(ensembl) %>%
  head(20)
attributeNames <- c('ensembl_gene_id_version',
                    'ensembl_gene_id',
                    'external_gene_name')
annot <- getBM(
  attributes = attributeNames,
  filters = filterType,
  values = filterValues,
  mart = ensembl
)
attributeNames <- c('ensembl_gene_id_version',
                    'gene_biotype',
                    'external_gene_name',
                    'description')
mart <- getBM(
  attributes = attributeNames,
  filters = filterType,
  values = filterValues,
  mart = ensembl
)
write_delim(
  annot,
  file = "assets/annot.tsv",
  delim = "\t"
)
write_delim(
  mart,
  file = "assets/mart.tsv",
  delim = "\t"
)    
```

Files were saved, so we don't query BioMart everytime we run the markdown. The files will be loaded, instead

```{r martObj2, warnings=TRUE, message=FALSE}
mart <-
    read.csv(
        "assets/mart.tsv",
        sep = "\t",
        header = TRUE,
    )
annot <-
    read.csv(
        "assets/annot.tsv",
        sep = "\t",
        header = TRUE,
    )
```


# Results

## Write files

```{r processDE2, warnings=TRUE, message=FALSE}
processDE.ver(res.KDvsCT, "KDvsCT")
```


## Volcano plots {.tabset}

### Control vs. KD (fc)

```{r vol1, fig.cap="Fig X: KD vs CT (purple/negFC = overexpressed in CT; green/posFC = overexpressed in KD)", fig.width=8, fig.height=5, warnings=FALSE, message=FALSE}

g <- volcanoPlots(
  res.se = res.KDvsCT,
  string = "KDvsCT",
  first = "Control",
  second = "KD",
  color1 = "#00b462",
  color2 = "#4d4d4d",
  color3 = "#980c80",
  ChartTitle = "Control vs. KD",
  labelType = "fc"
)
g + ylim(0,325)
```

### Control vs. KD (interactive)

```{r vol2, fig.cap="Fig X: KD vs CT (purple/negFC = overexpressed in CT; green/posFC = overexpressed in KD)", fig.width=8, fig.height=5, warnings=FALSE, message=FALSE}
ggplotly(g)
```

### Control vs. KD (padj)

```{r vol3, fig.cap="Fig X: KD vs CT (purple/negFC = overexpressed in CT; green/posFC = overexpressed in KD)", fig.width=8, fig.height=5, warnings=FALSE, message=FALSE}

g <- volcanoPlots(
  res.se = res.KDvsCT,
  string = "KDvsCT",
  first = "Control",
  second = "KD",
  color1 = "#00b462",
  color2 = "#4d4d4d",
  color3 = "#980c80",
  ChartTitle = "Control vs. KD",
  labelType = "padj"
)
g
```

## `enrichR`

```{r enrichR, warnings=TRUE, message=FALSE}
setEnrichrSite("Enrichr")
websiteLive <- TRUE
myDBs <-
  c(
    "DisGeNET",
    "GO_Biological_Process_2021",
    "HDSigDB_Human_2021",
    "KEGG_2021_Human",
    "GTEx_Tissues_V8_2023",
    "MGI_Mammalian_Phenotype_Level_3",
    "MGI_Mammalian_Phenotype_Level_4_2021",
    "WikiPathways_2019_Human",
    "Panther_2015"
  )
if (websiteLive) {
  KDvsCT.up.enriched <- enrichr(KDvsCT.up.pce2, myDBs)
  KDvsCT.dw.enriched <- enrichr(KDvsCT.dw.pce2, myDBs)
}
```

## PCE tests (green/posFC = overexpressed in KD) {.tabset}


### VentoTormo
```{r pceA1, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.up.pce1,  dataset = "Vt", myColor = "#00B462")
```

### Xiang
```{r pceA2, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.up.pce2, dataset = "Xi", myColor = "#00B462")
```

### Zhou&Petropoulos (Castel)

```{r pceA3, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.up.pce2, dataset = "Zp", myColor = "#00B462")
```

### Rostovskaya (Hs)

```{r pceA4, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.up.pce2, dataset = "Ro.Hs", myColor = "#00B462")
```

### Rostovskaya (Cy)

```{r pceA5, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.up.pce2, dataset = "Ro.Cy", myColor = "#00B462")
```


## PCE tests (purple/negFC = overexpressed in CT) {.tabset}

### VentoTormo
```{r pceB1, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.dw.pce1,  dataset = "Vt", myColor = "#980C80")
```

### Xiang
```{r pceB2, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.dw.pce2, dataset = "Xi", myColor = "#980C80")
```

### Zhou&Petropoulos (Castel)

```{r pceB3, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.dw.pce2, dataset = "Zp", myColor = "#980C80")
```

### Rostovskaya (Hs)

```{r pceB4, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.dw.pce2, dataset = "Ro.Hs", myColor = "#980C80")
```

### Rostovskaya (Cy)

```{r pceB5, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotPCE(inputGenes = KDvsCT.dw.pce2, dataset = "Ro.Cy", myColor = "#980C80")
```


## Enrichment tests (green/posFC = overexpressed in KD) {.tabset}


### DisGeNET
```{r erA1, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.up.enriched, table="DisGeNET" , myColor = "#00B462")
```


### GO_Biological_Process_2021
```{r erA2, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.up.enriched, table="GO_Biological_Process_2021" , "#00B462")
```

### HDSigDB_Human_2021

```{r erA3, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.up.enriched, table="HDSigDB_Human_2021" , "#00B462")
```

### KEGG_2021_Human

```{r erA4, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.up.enriched, table="KEGG_2021_Human" , "#00B462")
```

### GTEx_Tissues_V8_2023

```{r erA5, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.up.enriched, table="GTEx_Tissues_V8_2023" , "#00B462")
```

### MGI_Mammalian_Phenotype_Level_3

```{r erA6, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.up.enriched, table="MGI_Mammalian_Phenotype_Level_3" , "#00B462")
```

### MGI_Mammalian_Phenotype_Level_4_2021

```{r erA7, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.up.enriched, table="MGI_Mammalian_Phenotype_Level_4_2021" , "#00B462")
```

### WikiPathways_2019_Human

```{r erA8, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.up.enriched, table="WikiPathways_2019_Human" , "#00B462")
```

### Panther_2015
   
```{r erA9, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.up.enriched, table="Panther_2015" , "#00B462")
```

 

## Enrichment tests (purple/negFC = overexpressed in CT) {.tabset}


### DisGeNET
```{r erB1, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.dw.enriched, table="DisGeNET" , myColor = "#980C80")
```


### GO_Biological_Process_2021
```{r erB2, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.dw.enriched, table="GO_Biological_Process_2021" , "#980C80")
```

### HDSigDB_Human_2021

```{r erB3, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.dw.enriched, table="HDSigDB_Human_2021" , "#980C80")
```

### KEGG_2021_Human

```{r erB4, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.dw.enriched, table="KEGG_2021_Human" , "#980C80")
```

### GTEx_Tissues_V8_2023

```{r erB5, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.dw.enriched, table="GTEx_Tissues_V8_2023" , "#980C80")
```

### MGI_Mammalian_Phenotype_Level_3

```{r erB6, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.dw.enriched, table="MGI_Mammalian_Phenotype_Level_3" , "#980C80")
```

### MGI_Mammalian_Phenotype_Level_4_2021

```{r erB7, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.dw.enriched, table="MGI_Mammalian_Phenotype_Level_4_2021" , "#980C80")
```

### WikiPathways_2019_Human

```{r erB8, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.dw.enriched, table="WikiPathways_2019_Human" , "#980C80")
```

### Panther_2015
   
```{r erB9, fig.cap="", fig.width=12, fig.height=5, warnings=FALSE, message=FALSE}
plotEnrichR(KDvsCT.dw.enriched, table="Panther_2015" , "#980C80")
```

# GSEA

## Prepare GeneList

```{r geneListGSEA}
filteredDE <- fread("assets/DE_KDvsCT_filtered.tsv")
original_gene_list <- filteredDE$log2FoldChange
names(original_gene_list) <- gsub("\\..*", "", filteredDE$Gene)
gene_list<-na.omit(original_gene_list)
gene_list = sort(gene_list, decreasing = TRUE)
```

## Run GSEA

```{r runGSEA, message = TRUE}
gse <- gseGO(
  geneList = gene_list,
  ont = "ALL",
  keyType = "ENSEMBL",
  minGSSize = 3,
  maxGSSize = 800,
  pvalueCutoff = 0.05,
  verbose = TRUE,
  OrgDb = org.Hs.eg.db,
  pAdjustMethod = "BH",
  eps = 0,
  exponent = 1
)
saveRDS(gse, file = "assets/gsea.rds")
```

## Results table

```{r resultsTable}
gse <- readRDS(file = "assets/gsea.rds")
gseTable <- as_tibble(gse)
DT::datatable(
  gseTable,
  filter = "top",
  options = list(pageLength = 10)) %>%
  DT::formatRound(c("enrichmentScore", "NES", "pvalue", "p.adjust", "qvalue"), 2) %>%
  DT::formatStyle("core_enrichment",
              "white-space" = "nowrap")
```

## Plots {.tabset}

### Blobplot plot

```{r blobplot, fig.width = 12, fig.height = 8}
dotplot(gse, showCategory=10, split=".sign") + facet_grid(.~.sign)
```

### Enrichment Map

For the first 15 categories

```{r enrichmentMap}
mapGO <- godata('org.Hs.eg.db', ont = "BP")
emaGSE <- pairwise_termsim(gse, method="Wang", semData = mapGO)
emapplot(emaGSE, showCategory = 15)
```


### Category Netplot

Showing 5 categories

```{r netplot}
cnetplot(gse, categorySize="pvalue", foldChange=gene_list, showCategory = 5)
```


### Ridgeplot


```{r ridgeplot, fig.width = 12, fig.height = 8}
ridgeplot(gse) + labs(x = "enrichment distribution")
```

## GSEA plot {.tabset}

For the 4 categories

### set1

```{r gseaP1}
gseaplot(gse, by = "all", title = gse$Description[1], geneSetID = 1)
```

### set2

```{r gseaP2}
gseaplot(gse, by = "all", title = gse$Description[2], geneSetID = 2)
```

### set3

```{r gseaP3}
gseaplot(gse, by = "all", title = gse$Description[3], geneSetID = 3)
```

### set4

```{r gseaP4}
gseaplot(gse, by = "all", title = gse$Description[4], geneSetID = 4)
```


# KEGG

## Prepare GeneList

```{r geneListKEGG}
ids <- bitr(
  names(original_gene_list),
  fromType = "ENSEMBL",
  toType = "ENTREZID",
  OrgDb = org.Hs.eg.db
)
dedup_ids = ids[!duplicated(ids[c("ENSEMBL")]),]
filteredDE$Gene <- gsub("\\..*", "", filteredDE$Gene)
filteredDE2 <- left_join(dedup_ids, 
                         filteredDE, 
                         by = c('ENSEMBL' = 'Gene'))
kegg_gene_list <- filteredDE2$log2FoldChange
names(kegg_gene_list) <- filteredDE2$ENTREZID
kegg_gene_list <- na.omit(kegg_gene_list)
kegg_gene_list = sort(kegg_gene_list, 
                      decreasing = TRUE)
```

## run KEGG

```{r runKegg, message = TRUE}
orgCode <- "hsa"
keggPW <-gseKEGG(
  kegg_gene_list,
  organism = orgCode,
  keyType = "ncbi-geneid",
  exponent = 1,
  minGSSize = 3,
  maxGSSize = 800,
  eps = 1e-10,
  pvalueCutoff = 0.05,
  pAdjustMethod = "BH",
  verbose = TRUE,
  by = "fgsea")
saveRDS(keggPW, file = "assets/kegg.rds")
```

No terms enriched

# Save RData

Saving the entire session as well as the DEseq2 object (as `rds`) for doing the overlap analyses.

```{r saveInfo}
save.image(file = "assets/BeWo_PLAC1_KD.RData")
save(KDvsCT.up.pce1, KDvsCT.dw.pce1, file = "assets/BeWo_PLAC1_KD_genelists.RData")
```

# MultiQC report:

**NOT YET** MultiQC report is available at this [link](assets/multiqc_report.html){target="_blank"}


# Session Information

```{r sessioninfo}
sessionInfo()
```